### ISSUE TITLE
tensorflow/core/framework/common_shape_fns.h not in includes path (v0.11.0rc0)

### ISSUE THREAD

**Issue by sirfz** [tensorflow/tensorflow#4996]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
None

### Environment info
Operating System: **Ubuntu 14.04 x64**

Installed version of CUDA and cuDNN: **CUDA 7.5 and CUDNN 5.1**
```
$ ll /usr/local/cuda/lib64/libcud*
-rw-r--r-- 1 root root   322936 Aug 15  2015 /usr/local/cuda/lib64/libcudadevrt.a
lrwxrwxrwx 1 root root       16 Aug 15  2015 /usr/local/cuda/lib64/libcudart.so -> libcudart.so.7.5*
lrwxrwxrwx 1 root root       19 Aug 15  2015 /usr/local/cuda/lib64/libcudart.so.7.5 -> libcudart.so.7.5.18*
-rwxr-xr-x 1 root root   383336 Aug 15  2015 /usr/local/cuda/lib64/libcudart.so.7.5.18*
-rw-r--r-- 1 root root   720192 Aug 15  2015 /usr/local/cuda/lib64/libcudart_static.a
-rwxr-xr-x 1 root root 60696704 Sep 21 21:07 /usr/local/cuda/lib64/libcudnn.so*
-rwxr-xr-x 1 root root 60696704 Sep 21 21:07 /usr/local/cuda/lib64/libcudnn.so.5*
-rwxr-xr-x 1 root root 60696704 Sep 21 21:07 /usr/local/cuda/lib64/libcudnn.so.5.1.3*
-rw-r--r-- 1 root root 59715990 Sep 21 21:07 /usr/local/cuda/lib64/libcudnn_static.a
```

If installed from binary pip package, provide:

1. A link to the pip package you installed: **https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl**
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.
**0.11.0rc0**

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

Cannot `#include "tensorflow/core/framework/common_shape_fns.h` to use `::tensorflow::shape_inference::UnchangedShape`

Also:

```
$ ls /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/common_shape_fns.h
ls: cannot access /usr/local/lib/python2.7/dist-packages/tensorflow/include/tensorflow/core/framework/common_shape_fns.h: No such file or directory
```

### What other attempted solutions have you tried?

Use python to infer custom op's shape instead (which works):
```python
tf.RegisterShape("MyOp")(common_shapes.unchanged_shape)
```

**Comment by jart** [comment]
Thanks for reporting this issue. Can you provide commands to reproduce this problem?

**Comment by sirfz** [comment]
Here's a sample op (sample_op.cc):

```c++
#include "tensorflow/core/framework/op.h"
#include "tensorflow/core/framework/op_kernel.h"
#include "tensorflow/core/framework/shape_inference.h"
#include "tensorflow/core/framework/common_shape_fns.h"

using namespace tensorflow;

REGISTER_OP("SampleOp")
    .Input("input: string")
    .Output("output: string")
    /* .SetShapeFn([](shape_inference::InferenceContext* c) { */
    /*     shape_inference::ShapeHandle input; */
    /*     TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 1, &input)); */
    /*     if (c->Value(c->Dim(input, 0)) > 1) { */
    /*         return errors::InvalidArgument( */
    /*             "Shape of a default must be a length-0 or length-1 vector"); */
    /*     } */
    /*     c->set_output(0, input); */
    /*     return Status::OK(); */
    /* }) */
    .SetShapeFn([](shape_inference::UnchangedShape))
    .Doc(R"doc(
Return input strings uchanged
input: 1-D. Input strings
output: A vector of strings corresponding to the input strings.
)doc");

class SampleOp : public OpKernel {
 public:
  explicit SampleOp(OpKernelConstruction* context): OpKernel(context) {
  }

  void Compute(OpKernelContext* ctx) override {
    const Tensor* input_tensor;
    OP_REQUIRES_OK(ctx, ctx->input("input", &input_tensor));
    auto records_t = input_tensor->flat<string>();
    const int64 input_size = records_t.size();

    Tensor* output_t;
    OP_REQUIRES_OK(
        ctx, ctx->allocate_output(0, input_tensor->shape(), &output_t));
    auto output = output_t->flat<string>();

    for (int64 i = 0; i < input_size; ++i) {
      output(i) = records_t(i);
    }
  }
};

REGISTER_KERNEL_BUILDER(Name("SampleOp").Device(DEVICE_CPU), SampleOp);
```
Compiling this file yields:
```
$ TF_INC=$(python -c 'import tensorflow as tf; print(tf.sysconfig.get_include())')
$ g++ -std=c++11 -shared sample_op.cc -o sample_op.so -fPIC -I $TF_INC -O2
sample_op.cc:4:56: fatal error: tensorflow/core/framework/common_shape_fns.h: No such file or directory
compilation terminated.
```

**Comment by asimshankar** [comment]
Thanks for the report @sirfz . I think this was an oversight. For now, could you work around it by manually copying the header file over? I will look into getting it fixed at tip and for the next release, if not sooner

**Comment by asimshankar** [comment]
Reopening since the attempted fix was rolled back in 8a0e81d81f88d49f49af969e9e7fe5807322931d

We'll attempt another fix soon

**Comment by dhiren-hamal** [comment]
So, how did you solve that problem? I have the same problem and I didn't see any header files inside either in tensorflow/core/framework directory.
